### PR TITLE
added ambient-link ability sdk

### PR DESCRIPTION
**Pull request by robinlehrmann**
| Q             | A
| ------------- | ---
| License       | MIT

Added ambient.link ability pack.

### CODE REVIEW

**Comment by Nyholm** [comment]
Thank you for this PR. 

Im not sure why you create a "pack"? Why just not create a recipe for `ambient-link/ability-sdk`?

**Comment by robinlehrmann** [comment]
> Thank you for this PR.
> 
> Im not sure why you create a "pack"? Why just not create a recipe for `ambient-link/ability-sdk`?

You are right. I changed the name and removed the pack

File: ambient-link/ability-sdk/1.0/config/packages/ability.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+            - '%env(AMBIENT_LINK_MQTT_HOST)%'
+            - '%env(AMBIENT_LINK_PORT)%'
+            - '%env(AMBIENT_LINK_MONKEY_ID)%'
+
+    AmbientLink\SDK\\Mqtt\MqttAdapter:
+        arguments:
+            $username: '%env(AMBIENT_LINK_MQTT_USERNAME)%'
+            $password: '%env(AMBIENT_LINK_MQTT_PASSWORD)%'
+
+ability:
```
Comment (Nyholm) [inline ambient-link/ability-sdk/1.0/config/packages/ability.yaml:16]: This line and below can be removed.

File: ambient-link/ability-sdk/1.0/config/packages/ability.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
+services:
+    AmbientLink\SDK\SmartMonkeyConnection:
+        arguments: ['@ambient_link.connection_settings', '%env(AMBIENT_LINK_USERNAME)%', '%env(AMBIENT_LINK_PASSWORD)%']
```
Comment (Nyholm) [inline ambient-link/ability-sdk/1.0/config/packages/ability.yaml:3]: Dont forget to defined a service for `ambient_link.connection_settings`

File: ambient-link/ability-sdk/1.0/config/packages/ability.yaml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+#                    to: joinedRoom
+#                join_room:
+#                    from: leftRoom
+#                    to: joinedRoom
+#                leave_room:
+#                    from: joinedRoom
+#                    to: leftRoom
+#                leave_apartment:
+#                   from: leftRoom
+#                    to: notInAnyRoom
```
Comment (Nyholm) [inline ambient-link/ability-sdk/1.0/config/packages/ability.yaml:51]: I suggest removing lines 20 to 51. They dont help the user

File: ambient-link/ability-sdk/1.0/src/AbilityEventSubscriber.php [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+declare(strict_types=1);
+
+namespace AmbientLink\Contrib\Ability;
+
+use AmbientLink\SDK\Ability;
+use AmbientLink\SDK\Event\EnterRoomEvent;
+use Psr\Log\LoggerInterface;
+
+final class AbilityEventSubscriber implements Ability
```
Comment (Nyholm) [inline ambient-link/ability-sdk/1.0/src/AbilityEventSubscriber.php:11]: Is this class really needed? 

I think it can be removed.

**Review by Nyholm — CHANGES_REQUESTED** [review]
Thank you

File: ambient-link/ability-sdk/1.0/manifest.json [hunk 5]
```diff
@@ -0,0 +1,16 @@
+{
+    "env": {
+        "AMBIENT_LINK_MQTT_HOST": "",
+        "AMBIENT_LINK_PORT": "1883",
+        "AMBIENT_LINK_MQTT_USERNAME":  "",
+        "AMBIENT_LINK_MQTT_PASSWORD": "",
+        "AMBIENT_LINK_BASE_TOPIC": "",
```
Comment (Nyholm) [inline ambient-link/ability-sdk/1.0/manifest.json:7]: This is not used

File: ambient-link/ability-sdk/1.0/manifest.json [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+        "AMBIENT_LINK_MONKEY_ID":  "",
+        "AMBIENT_LINK_ABILITY_TOKEN": "",
```
Comment (Nyholm) [inline ambient-link/ability-sdk/1.0/manifest.json:9]: This is not used

File: ambient-link/ability-sdk/1.0/manifest.json [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+        "AMBIENT_LINK_MQTT_USERNAME_AND_PASSWORD": "",
```
Comment (Nyholm) [inline ambient-link/ability-sdk/1.0/manifest.json:10]: This is not used

File: ambient-link/ability-sdk/1.0/manifest.json [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+        "AMBIENT_LINK_MQTT_URL": "${MQTT_USERNAME_AND_PASSWORD}${MQTT_HOST}:1883"
```
Comment (Nyholm) [inline ambient-link/ability-sdk/1.0/manifest.json:11]: This is not used

File: ambient-link/ability-sdk/1.0/config/packages/ability.yaml [hunk 9]
```diff
@@ -0,0 +1,18 @@
+services:
+    AmbientLink\SDK\SmartMonkeyConnection:
+        arguments: ['@amnient.link_connection_settings', '%env(AMBIENT_LINK_USERNAME)%', '%env(AMBIENT_LINK_PASSWORD)%']
```
Comment (Nyholm) [inline ambient-link/ability-sdk/1.0/config/packages/ability.yaml:3]: There is no definition for `AMBIENT_LINK_USERNAME` or `AMBIENT_LINK_PASSWORD`

**Review by Nyholm — CHANGES_REQUESTED** [review]
Thank you. 

Can you make sure that you define all environment variables? Also, dont define more environment variables than you are using.

**Review by Nyholm — APPROVED** [review]
Thank you.

**Pull request merged**